### DISCUSSION TITLE
Any way to render directly to a window without WPF or WinForms?

### DISCUSSION THREAD

**Discussion by slxdy** [cefsharp/CefSharp#4699] [General]
Hey, I'm wondering if there is any way to make a very lightweight implementation of CefSharp directly to a window, without WPF or WinForms, possibly with just DirectX or OpenGL. Thx!

**Comment by amaitland** [comment]
> I'm wondering if there is any way to make a very lightweight implementation of CefSharp directly to a window,

What's a window in this context? A `Win32 Window`?

Generically speaking the core functionality is part of the public API and can be used to create your own specific implementation. Here are some examples.

https://github.com/cefsharp/CefSharp.OutOfProcess/blob/master/CefSharp.OutOfProcess.BrowserProcess/OutOfProcessChromiumWebBrowser.cs#L791
https://github.com/cefsharp/CefSharp.Wpf.HwndHost/blob/master/CefSharp.Wpf.HwndHost/ChromiumWebBrowser.cs#L748
https://github.com/cefsharp/CefSharp.Wpf.HwndHost/blob/master/CefSharp.Wpf.HwndHost/ChromiumWebBrowser.cs#L1544

You'll need a basic understanding of `CEF`. https://bitbucket.org/chromiumembedded/cef/wiki/GeneralUsage.md#markdown-header-application-structure is a good place to start.

> possibly with just DirectX or OpenGL

Not directly.
